sim/core/unit.go

```diff
@@ -110,6 +110,10 @@ type Unit struct {
 	CastSpeed float64
 
 	CurrentTarget *Unit
+
+	// DamgeTaken is used to track health fights instead of duration fights.
+	//  Once primary target has taken its health worth of damage, fight ends.
+	DamageTaken float64
```

Comment:
Unless you want to sum this for all targets, maybe this should be a sim field too alongside EndFightAtHealth?